### ISSUE TITLE
Only first value added to date time appears in rendered SQL

### ISSUE THREAD

**Issue by bradnewman** [jOOQ/jOOQ#9196]
### Expected behavior and actual behavior:
The sum of a date time and two or more other values renders only the date time and the first value.

e.g. `DSL.currentLocalDateTime().plus(1).plus(1).plus(1).toString()` yields: 
`"(current_timestamp() + 1)"`

The expected behavior is that all addends should be rendered, e.g. the above should instead yield
`"(current_timestamp() + 1 + 1 + 1)"`

This appears to be the result of a combination of [`Expression.add()` appending arguments to the `rhs`](https://github.com/jOOQ/jOOQ/blob/aaedefc9eb8303f27a68d8b06f704cfa434f7deb/jOOQ/src/main/java/org/jooq/impl/Expression.java#L139) and [`Expression.accept()` using a `DateExpression` with only `rhs.get(0)`](https://github.com/jOOQ/jOOQ/blob/aaedefc9eb8303f27a68d8b06f704cfa434f7deb/jOOQ/src/main/java/org/jooq/impl/Expression.java#L230).

The `minus()` and `sub()` methods are not affected by this issue, apparently because `Expression` does not override `sub()` as it does with `add()`, so the implementation from `AbstractField` is used instead.  Thus separate `Expression` instances with only one item in `rhs` are created for each subtraction.

jOOQ 3.11.12 works as expected, jOOQ 3.12.0 and 3.12.1 ignore the additional values.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

https://github.com/bradnewman/jOOQ-mcve

### Versions:

- jOOQ: Works as expected with 3.11.12; fails with 3.12.0 and 3.12.1.
- Java: 1.8.0_201 (but not relevant to the problem)
- Database (include vendor): PostgreSQL in prod, H2 for MCVE (but not relevant to the problem)
- OS: Linux x64 (but not relevant to the problem)
- JDBC Driver (include name if inofficial driver):

**Comment by knutwannheden** [comment]
Thank you very much for the detailed report (and the MCVE!).

We will look into this issue and include a fix in the next 3.12 patch release.

**Comment by lukaseder** [comment]
While the suggested fix would fix the problem by reverting a feature, I would still prefer to see if we can maintain the feature and still fix the problem exposed here (which is accessing only element `0` from the list, as observed by @bradnewman).

The feature is that:

```java
a.plus(b).plus(c)
```

... generates

```sql
-- This
(a + b + c)

-- Instead of this
((a + b) + c)
```

The current fix reverts to generating nested expressions. Notice we have the same behaviour with `Condition.and(..).and(..)` calls, which produce `(a and b and c)` instead of `((a and b) and c)`.

We could argue that the "feature" of not generating too many parentheses is a feature of arithmetic `+`, which is associative *and* commutative, whereas the `+` from date time arithmetic is not.
